src/weights.jl

```diff
@@ -53,9 +57,34 @@ Base.getindex(wv::W, ::Colon) where {W <: AbstractWeights} = W(copy(wv.values),
     isfinite(sum) || throw(ArgumentError("weights cannot contain Inf or NaN values"))
     wv.values[i] = v
     wv.sum = sum
+    wv.positive = missing
     v
 end
 
+function Base.all(f::Base.Fix2{typeof(>=)}, wv::AbstractWeights)
+    if iszero(f.x)
+        if ismissing(wv.positive)
```

Comment:
In my experience, in particular with `nothing`, it is often better to use `===` instead of `isnothing`/`ismissing` (see e.g. https://github.com/JuliaLang/julia/issues/35585#issuecomment-619053871):
```suggestion
        if wv.positive === missing
```

Reply:
Right. Though that's supposed to be better in recent releases, even if not completely conclusive (see https://github.com/JuliaLang/julia/pull/44407). Anyway that shouldn't matter here as we're not in a loop. But why not.